src/Microsoft.AspNetCore.ResponseCaching/KeyProvider.cs

```diff
@@ -153,5 +151,28 @@ public virtual string CreateVaryKey(HttpContext httpContext, VaryRules varyRules
                 _builderPool.Return(builder);
             }
         }
+
+        public virtual string CreateBodyKey(HttpContext httpContext)
```

Comment:
Looking into sharding the body in a separate branch. The logic here may eventually take in a chunk number as well to generate the key for a specific shard.